### PR TITLE
[ALLUXIO-2567]Append commons-codec dependency in alluxio-underfs-oss module.

### PR DESCRIPTION
**Pull request by maobaolong**
(https://alluxio.atlassian.net/browse/ALLUXIO-2567)

### CODE REVIEW

File: underfs/oss/pom.xml [hunk 1]
```diff
@@ -43,6 +43,10 @@
       <groupId>org.apache.httpcomponents</groupId>
       <artifactId>httpcore</artifactId>
     </dependency>
+    <dependency>
```
Comment (jsimsa) [inline underfs/oss/pom.xml]: please keep the external dependencies sorted lexicographically ... so move this above `httpcore`

**Comment by calvinjia** [comment]
@maobaolong Thanks for the contribution, could you open a JIRA ticket with more details on this issue? I would like to understand why this would fail only some of the time.

File: underfs/oss/pom.xml (hunk 1 above)
Comment (maobaolong) [inline underfs/oss/pom.xml]: Done.

**Comment by maobaolong** [comment]
@calvinjia , PTAL. [ALLUXIO-2567](https://alluxio.atlassian.net/browse/ALLUXIO-2567)

**Comment by calvinjia** [comment]
@maobaolong Thanks, let's continue the discussion on the JIRA ticket.

**Comment by maobaolong** [comment]
@calvinjia PTAL. [ALLUXIO-2567](https://alluxio.atlassian.net/browse/ALLUXIO-2567)

**Comment by yupeng9** [comment]
@calvinjia any updates on this PR?

**Comment by calvinjia** [comment]
@yupeng9 I'm discussing with @maobaolong on the JIRA ticket, it seems like this issue only occurs in a certain environment, so we are seeing if it is necessary to include this change as commons-codec should be included as a dependency of the OSS-SDK.

**Comment by maobaolong** [comment]
@calvinjia Sorry, I cannot access jira correctly.
wx_camera_1489557132780
wx_camera_1489557097775

**Comment by calvinjia** [comment]
@maobaolong Oh okay, let me cross post what I said in JIRA:

Thanks for the info. It looks like this is the culprit:
```
[WARNING] The POM for org.apache.httpcomponents:httpclient:jar:4.4 is invalid, transitive dependencies (if any) will not be available, enable debug logging for more details
```

Could you try finding out why that pom would be invalid in your environment?

**Comment by calvinjia** [comment]
@maobaolong Were you able to look into this?

**Comment by yupeng9** [comment]
@maobaolong any update on this?

**Comment by maobaolong** [comment]
@calvinjia  @yupeng9  Sorry for the missing days. I think i'd better to attach the new maven build log after add `-X` and `-Dgmaven.logging=DEBUG` arguments.

The following is the full log with those arguments to get more log.

[mvn.txt](https://github.com/Alluxio/alluxio/files/902655/mvn.txt)

**Pull request closed without merging**